### PR TITLE
Add beforeDragStop callback.

### PR DESCRIPTION
**Pull request by RomanBurunkov**
Add beforeDragStop callback function.

If function returns false, drag&drop will be canceled and list comes back to start state.

### CODE REVIEW

**Comment by DVD27** [comment]
This doesn't work well.

If you create a liste like this : 

- BarbaPapa
- - BarbaTruc

Now create a function beforeDragStop like  : 
`function beforeDragStop(list, elt, parent) {
    return false;
}`

Drag the element BarbaTruc to the same level than BarbaPapa (so in root)

Your element is not correctly rollback because the list doesn't exist anymore.

Excuse for my poor english.

File: jquery.nestable.js [hunk 1]
```diff
@@ -662,42 +663,59 @@
...
-                    $(currentEl.children[indexArray[indexArray.length - 1] - 1]).after(dragElement.clone());
+            }
+            //Diggin through indexArray to get home for dragElement.
+            for(var i = 0; i < indexArray.length; i++) {
+                if(lastIndex === parseInt(i)) {
+                    placeElement(currentEl, dragElement);
+                    return;
                 }
+                //element can have no indexes, so we have to use conditional here to avoid errors.
+                currentEl = (currentEl[0]) ? currentEl[0].children[indexArray[i]] : currentEl.children[indexArray[i]];
```
Comment (DVD27) [inline jquery.nestable.js]: We could fix the probleme by change this line in : 

```
var nextEl = (currentEl[0]) ? currentEl[0].children[indexArray[i]] : currentEl.children[indexArray[i]];
if (!nextEl) {
    // Rebuild the list if it missing
    var prev = (currentEl[0]) ? currentEl[0] : currentEl;
    var list = $('<' + this.options.listNodeName + '/>').addClass(this.options.listClass);
    prev.append(list[0]);
    this.setParent($(prev));
    currentEl = list;
} else {
    currentEl = nextEl;
}
```

**Comment by RomanBurunkov** [comment]
Hi @DVD27 , thanks for this feedback and patch proposal.

Going to check it this evening(GMT+3).

**Comment by RomanBurunkov** [comment]
Hi,
I've updated PL with corrections to bug noticed by @DVD27.

Thanks @DVD27  ;)

File: jquery.nestable.js [hunk 2]
```diff
@@ -56,6 +56,7 @@
         includeContent: false,
         callback: function(l, e, p) {},
         onDragStart: function(l, e, p) {},
+        beforeDragStop: function(l, e, p) {},
```
Comment (pjona) [inline jquery.nestable.js]: We should also add information about new function in `README.md`

**Comment by RomanBurunkov** [comment]
Hi @pjona ,

It looks like sublevel is adding at restoreItemAtIndex, checking how to solve this.

**Comment by RomanBurunkov** [comment]
@pjona ,
I've added check for setParent to make sure that target element has itemNodeName type.
This is simplest way to solve this.

**Comment by RomanBurunkov** [comment]
I've added information about beforeDragStop to `README.md`.

File: jquery.nestable.js [hunk 3]
```diff
@@ -662,42 +675,67 @@
...
-                    $(currentEl.children[indexArray[indexArray.length - 1] - 1]).after(dragElement.clone());
+            }
+            //Diggin through indexArray to get home for dragElement.
+            for(var i = 0; i < indexArray.length; i++) {
+                if(lastIndex === parseInt(i)) {
+                    placeElement(currentEl, dragElement);
+                    return;
                 }
+                //element can have no indexes, so we have to use conditional here to avoid errors.
+                //element can have no indexes, so we have to use conditional here to avoid errors.
```
Comment (pjona) [inline jquery.nestable.js]: This comment is duplicated :)

Comment (RomanBurunkov) [inline jquery.nestable.js]: Removed =)

**Comment by pjona** [comment]
@RamonSmit can you make review?

**Pull request merged**